src/reference/asciidoc/mqtt.adoc

```diff
@@ -500,4 +500,65 @@ IMPORTANT: The `org.springframework.integration.mqtt.support.MqttMessageConverte
\ No newline at end of file
+Otherwise, only the manual restart of `Mqttv5PahoMessageDrivenChannelAdapter` can handle reconnects, e.g. via `MqttConnectionFailedEvent` handling on disconnection.
+
+[[mqtt-shared-client]]
+=== Shared MQTT Client Support
+
+If you need to use a single MQTT ClientID for several integrations, you cannot use different MQTT client instances because MQTT brokers may have a limitation on a number of connections per ClientID (typically, a single connection is allowed).
+For having a single client reused for different channel adapters, you may use a `org.springframework.integration.mqtt.core.ClientManager` component and pass it to any channel adapter you need.
+It will manage MQTT connection lifecycle and do automatic reconnects if needed.
+You can also provide a custom connection options and `MqttClientPersistence` just as you would do it for channel adapter components.
+
+Note that both MQTT v5 and v3 channel adapters are supported.
+
+The following Java DSL configuration sample demonstrates how to use this client manager in the integration flow:
+
+====
+[source,java]
+----
+@Bean
+public ClientManager<IMqttAsyncClient, MqttConnectionOptions> clientManager() {
+    MqttConnectionOptions connectionOptions = new MqttConnectionOptions();
+    connectionOptions.setServerURIs(new String[]{ "tcp://localhost:1883" });
+    connectionOptions.setConnectionTimeout(30000);
+    connectionOptions.setMaxReconnectDelay(1000);
+    connectionOptions.setAutomaticReconnect(true);
+    Mqttv5ClientManager clientManager = new Mqttv5ClientManager(connectionOptions, "client-manager-client-id-v5");
+    clientManager.setPersistence(new MqttDefaultFilePersistence());
+    return clientManager;
+}
+
+@Bean
+public IntegrationFlow mqttInFlowFoo(
+        ClientManager<IMqttAsyncClient, MqttConnectionOptions> clientManager) {
+
+    Mqttv5PahoMessageDrivenChannelAdapter messageProducer =
+        new Mqttv5PahoMessageDrivenChannelAdapter(clientManager, "foo-topic");
+    return IntegrationFlow.from(messageProducer)
+            .channel(c -> c.queue("fromMqttChannel"))
+            .get();
+}
+
+@Bean
+public IntegrationFlow mqttInFlowBar(
+        ClientManager<IMqttAsyncClient, MqttConnectionOptions> clientManager) {
+
+    Mqttv5PahoMessageDrivenChannelAdapter messageProducer =
+        new Mqttv5PahoMessageDrivenChannelAdapter(clientManager, "bar-topic");
+    return IntegrationFlow.from(messageProducer)
+            .channel(c -> c.queue("fromMqttChannel"))
+            .get();
+}
+
+@Bean
+public IntegrationFlow mqttOutFlow(
+        ClientManager<IMqttAsyncClient, MqttConnectionOptions> clientManager) {
+
+    return f -> f.handle(new Mqttv5PahoMessageHandler(clientManager));
+}
+----
+====
+
+IMPORTANT: You should use a single client manager instance so that the connection is reused.
+Ensure you do not create it for each channel adapter.
```

Comment:
I don't think we need this `IMPORTANT`. Not sure how it is important at all since client per channel adapter is still a valid use-case.
Please, consider to remove this altogether.

A couple notes about technical docs:

1. It is better to express the matter in the impersonal manner. In other words we must not say `you`, `we`, `they` or so. I mean not `you cannot use different`, but `the same MQTT client id cannot be used`. 
2. `Foo` and `bar` are not professional words and are considered as a historical mistake for software engineering practice. So, better to call them `topic1` and `topic2`.


Both of these are optional at the moment and feel free to pass such a clean up task up to me.
Just let me know your thoughts!

Reply:
These all make sense, I have changed it.